src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -288,7 +288,6 @@ private static void invokeWithTimeoutWithNewExecutor(ITestNGMethod tm, Object in
       ThreadTimeoutException exception = new ThreadTimeoutException("Method "
           + tm.getQualifiedName() + "()"
           + " didn't finish within the time-out " + realTimeOut);
-      exception.setStackTrace(exec.getStackTraces()[0]);

```

Comment:
`ThreadUtil.createExecutor(1, tm.getMethodName());` can explain why we took only the first element of the array (a comment missed before).

But it doesn't explain why `ThreadFactoryImpl` doesn't feed its `threads` list.
https://github.com/cbeust/testng/blob/15a824bd8d3658e7070b7cf105f2adc011d07000/src/main/java/org/testng/internal/thread/ThreadUtil.java#L90
I can't remember why I didn't keep it in https://github.com/cbeust/testng/commit/1b8bd784ee975cf3983ebb7c12eb539985720fbb#diff-a8556e01a426efbe66d3b76a06249c04L108
I guess the issue will be fixed if you restore the thread list usage :)